### ISSUE TITLE
JOIN in INSTANCE and multiple JOINS

### ISSUE THREAD

**Issue by lmichel** [ivoa-std/ModelInstanceInVot#76]
The case 8.20 validates the following pattern:
```xml
<dm-mapping:COLLECTION dmrole="iiiiii">
	<dm-mapping:INSTANCE dmtype="model:object" />
	<dm-mapping:JOIN tableref="_someTable" dmref="_extInst1"/>
	<dm-mapping:JOIN tableref="_someOtherTable" dmref="_extInst2"/>
	<dm-mapping:INSTANCE dmtype="model:object" />
	<dm-mapping:INSTANCE dmtype="model:object" />
</dm-mapping:COLLECTION>
```

I do not see how a client could process that double JOIN!
JOIN are meant to return a set of items: they must be enclosed in a COLLECTION, isn’t it?
This pattern should be rejected

**Comment by lmichel** [comment]
The following pattern is accepted so far:

```
GLOBALS
   COLLECTION
      JOIN
       INSTANCE
       INSTANCE
```
In this case the collection is populated with a set of INSTANCE pairs populated with the JOIN rules.
This shouldn’t be allowed since we do not support collections as sets of 2uples.

**Comment by lmichel** [comment]
I propose to modify the schema to only support the following patterns (WHERE skipped)

```
COLLECTION
    JOIN @tableref
```
collection populated with mapped table rows

```
COLLECTION
    JOIN @tableref
    INSTANCE (only one)
```
collection populated with table data mapped on INSTANCE, one instance per collection item

```
COLLECTION
    JOIN @dmref
```
collection populated with items of the static COLLECTION dmid=dmref

**Comment by lmichel** [comment]
I do not remind the case where you wished to keep `<JOIN @dmref @tablerf>`
I would be happy to ban this over-statement, except if you insist to keep it

**Comment by lmichel** [comment]
The rule is that the mapping of the joined data is always set in the target (joined) table or collection.
It cannot be set within the JOIN statement.

**Comment by mcdittmar** [comment]
Case (8.20) is showing a COLLECTION which has contributors from different sources.  The JOINs are to be executed in sequence, each adding to the full COLLECTION.
   1. 3 locally defined INSTANCES (content is in the current TABLE)
   2. a set of instances pulled from the TABLE  "_someTable", the INSTANCE description is at _dmExtInst1 (within the annotation for that TABLE)
   3. a set of instances pulled from the TABLE "_someOtherTable", the INSTANCE description is at _dmExtInst2 (within the annotation for that TABLE).

> The following pattern is accepted so far:
```
GLOBALS
   COLLECTION
      JOIN
       INSTANCE
       INSTANCE
```
I cannot tell if the indentation here is correct.  Join does not currently allow INSTANCE as a child, nor should it.
Since the INSTANCE itself is in a different table from the JOIN tag, it cannot be defined within the JOIN (the column references would not match) 

Bottom line:
Is your proposal that COLLECTION with JOIN child
  * has ONLY JOIN child element.. ie: that INSTANCE is no longer allowed as well.
  * that there is only 1 JOIN child allowed

I seem to recall that this was targeted toward something.
From a general perspective, the current option (COLLECTION of JOINed and local INSTANCES) seems reasonable.
But, without a specific case, and knowing it could be supported.. I don't have an objection to tightening the constraints there.

**Comment by lmichel** [comment]
IN the example you give abovem you should have:
```
COLLECTION
  INSTANCE
  INSTANCE
  INSTANCE
  JOIN
  JOIN
```
After having struggled with JOIN in my code, I would be relieved not to have to populate collection with items from different sources and to limit the number of COLLECTION patterns
```
COLLECTION
  INSTANCE/ATTRIBUTE/REFERENCE/COLLECTION 0-*
```
or
```
COLLECTION
  JOIN
```
